### PR TITLE
Rename Const to Enum

### PR DESCRIPTION
**Pull request by stonechoe**
This PR is a refactor designed to implement the changes that were agreed upon during the documentation process for ESMeta.

###### Done
* Rename `Const` to `Enum` to align with the terminology used in [ECMA-262](https://tc39.es/ecma262/?#sec-enum-specification-type).

### CODE REVIEW

**Review by jhnaldo — APPROVED** [review]
LGTM! 👍🏻 I will merge this PR after passing all tests.

**Comment by jhnaldo** [comment]
@stonechoe Can you change the title of this PR before merging it? The original title is better than the current one.

**Comment by stonechoe** [comment]
> @stonechoe Can you change the title of this PR before merging it? The original title is better than the current one.

Done!

**Pull request merged**
